### ISSUE TITLE
automatic hyperparameter tuning

### ISSUE THREAD

**Issue by jeff-regier** [YosefLab/scVI#258]
It'd be nice if scVI would find good settings for it's hyperparameters automatically.

https://github.com/hyperopt/hyperopt

**Comment by jules-samaran** [comment]
@jeff-regier  Would this be the list of hyperparameters we would like to optimize automatically before training?
- number of epochs of training
- dimension of the latent space
- number of nodes per hidden layer
- number of layers used for encoder and decoder NNs
- dropout rate for neural networks
- weight of the kl divergence in the loss
- train size (proportion of the original dataset we train on)
- learning rate
Among all the parameters I cited some might have a predefined value we might want to use because we know it's reasonable rather than searching over all these parameters

**Comment by jeff-regier** [comment]
That's a pretty good list!
